src/ssd/ssd-part.c

```diff
@@ -142,18 +143,31 @@ get_scale_box(struct wlr_buffer *buffer, double container_width,
 
 struct ssd_part *
 add_scene_button(struct wl_list *part_list, enum ssd_part_type type,
-		struct wlr_scene_tree *parent, float *bg_color,
-		struct wlr_buffer *icon_buffer, struct wlr_buffer *hover_buffer,
-		int x, struct view *view)
+		enum ssd_part_type corner_type, struct wlr_scene_tree *parent,
+		float *bg_color, struct wlr_buffer *icon_buffer,
+		struct wlr_buffer *hover_buffer, int x, struct view *view)
 {
 	struct ssd_part *button_root = add_scene_part(part_list, type);
 	parent = wlr_scene_tree_create(parent);
 	button_root->node = &parent->node;
 	wlr_scene_node_set_position(button_root->node, x, 0);
 
+	int offset_x = 0;
+	if (corner_type == LAB_SSD_PART_CORNER_TOP_LEFT) {
+		offset_x = rc.theme->border_width;
+	} else if (corner_type == LAB_SSD_PART_CORNER_TOP_RIGHT) {
+		offset_x = 0;
+	} else {
+		assert(false && "invalid corner button type");
+		wlr_log(WLR_ERROR, "invalid corner button type");
+		abort();
+	}
+
 	/* Background */
 	struct ssd_part *bg_rect = add_scene_rect(part_list, type, parent,
-		SSD_BUTTON_WIDTH, rc.theme->title_height, 0, 0, bg_color);
```

Comment:
mmh, just resizing this part to include the borders isn't enough. Looks fine when the view is activated, but the boarder color is not correct when the view is not activated. That is with my theme.

Reply:
Is it because the button are positioned at intervals of `SSD_BUTTON_WIDTH`? At a glance, I'm guessing that if you make the buffer wider it will probably overlap adjacent buttons.

https://github.com/labwc/labwc/blob/1495708095b998f03aa8f07a6796764981e2dcd6/src/ssd/ssd-titlebar.c#L203-L228

Reply:
Thanks for the hint. I don't (yet) understand exactly what you mean, but mostly because all SSD painting is still magic to me ;). 

Here is a screenshot that shows what I mean:

![screenshot](https://github.com/labwc/labwc/assets/17219293/6e707192-d0e2-4cc7-8226-95fd257ac05b)

These are both SSD borders, both tiled and the left view (gnome-terminal) has the focus. The corner from gnome-terminal (left) looks fine, but the button on the CLion SSD doesn't shows the thin border line. I don't know yet where the thin boarder comes from. I do think though, that by making the button background bigger, the buttons are now on top of it and my theme colors are masking this when the view is activated.

Reply:
After discussion on chat, this should be good now and it's also much cleaner code-wise.